### PR TITLE
[job submission] Better validation + tests for input types, refactor API

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

- Hand-rolled validation (because we don't use FastAPI) :(
- Reworks the API to be match canon before public launch

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/utils.py [hunk 1]
```diff
@@ -128,11 +128,10 @@ async def _handler_route(*args) -> aiohttp.web.Response:
                     #   * (self, Request)
                     req = args[-1]
                     return await handler(bind_info.instance, req)
-                except Exception as e:
```
Comment (jiaodong) [inline dashboard/utils.py]: job_head part makes sense to me, but why is this needed ?

Comment (edoakes) [reply dashboard/utils.py]: Ah I reverted it, but this would get rid of the insane behavior where we return a 200 when we have an exception…

Comment (edoakes) [reply dashboard/utils.py]: We should do it sometime soon but it broke tests

**Review by jiaodong — APPROVED** [review]
